compute-hypothetical-feature.mjs

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env node
+import {readFileSync} from 'node:fs';
+
+import {compare} from 'compare-versions';
+
+const caniuse = JSON.parse(readFileSync('caniuse/fulldata-json/data-2.0.json', 'utf8'));
+const historicalBrowserData = JSON.parse(readFileSync('historical-browser-data.json', 'utf8'));
+
+// Sort the data in |historicalBrowserData| so that we can search it by date later.
+for (const snapshots of Object.values(historicalBrowserData)) {
+    snapshots.sort((a, b) => a.timestamp - b.timestamp);
+}
+
+const browsers = ['chrome', 'edge', 'firefox', 'safari'];
+
+const endDate = new Date();
+
+// Pre-process |a| and |b| to normalize some caniuse versions.
+function caniuseCompare(a, b, cmp) {
+    function normalize(version) {
+        if (version == 'TP') {
+            // Treat Safari TP as a future release. It doesn't matter because
+            // it has no usage in the caniuse data.
+            return String(Number.MAX_SAFE_INTEGER);
+        }
+        if (version == '15.2-15.3') {
+            return '15.2';
+        }
+        return version;
+    }
+    return compare(normalize(a), normalize(b), cmp);
+}
+
+// timestamp is number of seconds since epoch
+function caniuseDate(timestamp) {
+  return new Date(1000 * timestamp);
+}
+
+// from (inclusive) and to (exclusive) are Date objects.
+// TODO: this isn't quite right, date 
+function* iterateMonths(from, to) {
+    const iter = new Date(from);
+    while (iter < to) {
+        yield new Date(iter);
+        // Add one calendar month.
+        const nextMonth = (iter.getUTCMonth() + 1) % 12;
+        if (nextMonth == 0) {
+            iter.setUTCFullYear(iter.getUTCFullYear() + 1);
+        }
+        iter.setUTCMonth(nextMonth);
+    }
+}
+
+function monthString(date) {
+    return date.toISOString().substr(0, 10);
+}
+
+// Find the usage of a browser at a given date. The exact date is probably not
+// found in |historicalBrowserData|, so use the latest entry at or before
+// |date|. If there is none, return the very latest entry. The caller needs to
+// check that the data is fresh enough before using it.
+function getHistoricalUsage(browser, date) {
+    const snapshots = historicalBrowserData[browser];
+    // TODO: binary search would be O(log(n)) instead of O(n).
+    const index = snapshots.findLastIndex((s) => caniuseDate(s.timestamp) <= date);
```

Comment:
@dfabulich I found that the timestamps in historical-browser-data.json are strings, so I've sent https://github.com/dfabulich/baseline-calculator/pull/6. It's masked in this PR by always using it as a number, but it tripped me up at first when trying just `new Date(timestamp)` (which would have been wrong, but anyway).